test.sh

```diff
@@ -530,70 +566,138 @@ function test_make_run_tests() {
 
 # ##################################################################
 # Smoke Tests: Run a small collection of fast-running unit-tests
+# This can be invoked w/ or w/o the "--use-shmem" arg.
 # ##################################################################
 function run_fast_unit_tests() {
-
-   "$BINDIR"/unit/splinterdb_quick_test
-   "$BINDIR"/unit/btree_test
-   "$BINDIR"/unit/util_test
-   "$BINDIR"/unit/misc_test
-   "$BINDIR"/unit/limitations_test
-   "$BINDIR"/unit/task_system_test
-
-   "$BINDIR"/driver_test io_apis_test
+   local use_shmem=$1
+
+   "$BINDIR"/unit/splinterdb_quick_test "$use_shmem"
+   "$BINDIR"/unit/btree_test "$use_shmem"
+   "$BINDIR"/unit/util_test "$use_shmem"
+   "$BINDIR"/unit/misc_test "$use_shmem"
+   "$BINDIR"/unit/limitations_test "$use_shmem"
+   "$BINDIR"/unit/task_system_test "$use_shmem"
+
+   # Just exercise with some combination of background threads to ensure
+   # that basic usage of background threads still works.
+   # shellcheck disable=SC2086
+   "$BINDIR"/unit/task_system_test $use_shmem --num-bg-threads 4 --num-memtable-bg-threads  2
+
+   # shellcheck disable=SC2086
+   "$BINDIR"/driver_test io_apis_test $use_shmem
```

Comment:
All of these fast unit-tests can now be run with `--use-shmem` option.  You will get this coverage in every test run.